text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,589 @@
+- Feature Name: media-webrtc-sdk
+- Start Date: 2024-02-02
+- RFC PR: [8xff/rfcs#0000](https://github.com/8xff/rfcs/pull/0000)
+
+# 1. Summary
+
+[summary]: #summary
+
+This RFC provides an overview of how the WebRTC client connects to atm0s-media-server and its functionality.
+
+# 2. Motivation
+
+[motivation]: #motivation
+
+To enhance the user experience, we aim to develop a client SDK that is platform-agnostic and highly customizable. This SDK should provide a seamless integration with atm0s-media-server network topology and align with our approach to handle media streams. Our primary focus is to ensure that the SDK is user-friendly and easily adaptable to various use cases.
+
+# 3. User Benefit
+
+User can use WebRTC to connect to the media server and create complex media stream topologies.
+
+# 4. Design Proposal
+
+To ensure simplicity, we propose a SDK protocol that exclusively uses HTTP (not WebSocket) and WebRTC.
+
+- HTTP is used only for sending RPC requests to the cluster, typically during the Connect and Retry phases.
+- WebRTC is utilized for sending and receiving media streams, as well as RPC and event communication after establishing a connection.
+
+## 4.1 HTTP Request/Response format
+
+
+**Request and Response Format**
+
+All requests and responses will be encoded in JSON format. The format is described as follows:
+
+**_Body:_**: JSON
+
+**_Response:_**
+
+```
+{
+    success: bool,
+    error_code: Option<String>,
+    error_msg: Option<String>,
+    data: Option<JSON>,
+}
+```
+
+
+## 4.2 Connect Establishment
+
+Before connecting to the server, the client needs to prepare the following:
+
+- Enable WebRTC connection with data channel.
+- Create a list of senders and receivers.
```

Comment:
I think Senders and Receivers should be explained in more details about their purpose and design philosophy?